src/main/java/dev/j3fftw/litexpansion/items/WateringCan.java

```diff
@@ -0,0 +1,163 @@
+        return e -> {
+            e.cancel();
+            Optional<Block> bOpt = e.getClickedBlock();
+
+            if (bOpt.isPresent()) {
+                Block b = bOpt.get();
+                ItemStack item = e.getItem();
+                Player p = e.getPlayer();
+                BlockData blockData = b.getBlockData();
+                ItemMeta meta = item.getItemMeta();
+                List<String> lore = meta.getLore();
+
+                // Filling
+                if (b.getType() == Material.CAULDRON) {
+
+                    if (blockData.getAsString().contains("level=3")) {
+
+                        item.setType(Material.WATER_BUCKET);
+                        Utils.send(p, "&aYou have filled your Watering Can");
+                        lore.set(7, ChatColors.color("&aUses Left: &e5"));
+
+                    } else {
+                        Utils.send(p, "&cYou can only fill a Watering Can using a full cauldron!");
+                    }
+
+                    // Growing crops
+                } else {
+
+                    int uses = Integer.parseInt(String.valueOf(lore.get(USE_INDEX).charAt(15)));
+
+                    if (uses == 0) {
+                        Utils.send(p, "&cYou need to refill your Watering Can!");
+                        return;
+                    }
+
+                    if (b.getType() == Material.SUGAR_CANE) {
+
+                        int distance = 2;
+                        Block above = b.getRelative(BlockFace.UP);
+
+                        while (above.getType() == Material.SUGAR_CANE) {
+
+                            // Failsafe
+                            if (distance >= 10) {
+                                Utils.send(p, "&cThis sugar cane is too tall!");
+                                return;
+                            }
+
+                            above = b.getRelative(BlockFace.UP, distance);
+                            distance++;
+                        }
+
+                        if (above.getType() == Material.AIR) {
+
+                            above.setType(Material.SUGAR_CANE);
+                            uses--;
+
+                        } else {
+                            Utils.send(p, "&cThe sugar cane is obstructed!");
+                        }
+
+                    } else if (blockData instanceof Ageable) {
+
+                        Ageable crop = (Ageable) blockData;
+                        int currentAge = crop.getAge();
+                        int maxAge = crop.getMaximumAge();
+
+                        if (currentAge < maxAge) {
+                            crop.setAge(currentAge + 1);
+                            uses--;
+
+                        } else {
+                            Utils.send(p, "&cThis crop is already ready for harvest!");
+                        }
+
+                        b.setBlockData(blockData);
+                    } else if (b.getType().toString().endsWith("SAPLING")) {
```

Comment:
```suggestion
                    } else if (Tag.SAPLINGS.isTagged(b.getType()) {
```

Reply:
This is not resolved yet...

Reply:
changed, didnt push

Reply:
Please mark it resolved only after you pushed the change